scripts/stablex/place_margin_orders.js

```diff
@@ -0,0 +1,91 @@
+const BatchExchange = artifacts.require("BatchExchange")
+const { sendTxAndGetReturnValue } = require("../../test/utilities.js")
+const BN = require("bn.js")
+
+const argv = require("yargs")
+  .option("tokens", {
+    describe: "Collection of trusted tokens",
+  })
+  .option("accountId", {
+    describe: "Account index of the order placer",
+  })
+  .option("margin", {
+    type: "float",
+    describe: "Percentage increase required for trade (fees not accounted)",
+    default: 0.2,
+  })
+  .option("sellAmount", {
+    type: "float",
+    describe: "Maximum sell amount (considered infinite if empty)",
+    default: 100,
+  })
+  .option("expiry", {
+    type: "int",
+    describe: "Maximum auction batch for which these orders are valid",
+    default: 2**32 - 1
+  })
+  .demand(["tokens", "accountId"])
+  .help(false)
+  .version(false).argv
+
+
+// TODO - automate this with tokenIdToAddress dot decimals
```

Comment:
...or use our list as `scripts/stablex/add_token_list.js ` do

Reply:
If you do that, you can print nice messages with the orders in the console

Also, the for loop would loop a bit nicer, less "i" "j", etc.

Reply:
I would love to name the tokens via their symbols, not with the address.

Reply:
Yes I have already started this. But I consider referring to the token list as kind of a “half ass” solution. That also kinda requires users to “trust” we won’t change it. I am going with this option for now, because I have not been able to access the ‘decimals’ attribute of the ERC20 token for some reason

Reply:
> Also, the for loop would loop a bit nicer, less "i" "j", etc.

Not sure what you mean here, but I am not aware of any cleaner way to iterate over all distinct pairs in JS. If you know of anything like the following python code in JS I would be happy to hear:

```py
from itertools import combinations
for v, w in combinations(trusted_tokens, 2):
```

Reply:
For the list, you can count in the near future an API with all the tokens with their name, symbol, decimal and tokenId.

I'll let you know. For the time being, using the list sound reasonable

For the combinations JS has less fancy ways to do that, they have, but in libraries. For some reason I thought that we could iterate without those indexes, but it's good as it is.